tests/kernel/sleep/src/main.c

```diff
@@ -171,8 +175,12 @@ static void irq_offload_isr(const void *arg)
 	k_wakeup((k_tid_t) arg);
 }
 
-static void helper_thread(int arg1, int arg2)
+static void helper_thread(void *p1, void *p2, void *p3)
 {
+	ARG_UNUSED(p3);
+
+	int arg1 = p1;
+	int arg2 = p2;
```

Comment:
These will cause warnings, you have to cast. `POINTER_TO_INT(p1);`

Reply:
please review other cases like this one.

Reply:
CI was definitely running with `-Werror` the last time I checked.

Reply:
I went through all the changes and I think I found all these occasions.